### PR TITLE
Port testSuite/../java.math to JUnit and cross compile. 

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
@sjrd of all the "Port and cross-compile..." PR this is the first one you should look at.

**Comment by sjrd** [comment]
Why rename `AssertThrows` into `AssertThrowsUtil`?

**Comment by sjrd** [comment]
What's happening with the "No changes" in the files at the end of the first commit? What's wrong with them?

File: test-suite/js/src/test/require-jdk8/org/scalajs/testsuite/javalib/math/BigIntegerConvertJSTest.scala [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import java.math.BigInteger
+
+import org.scalajs.jasminetest.JasmineTest
+
+object BigIntegerConvertJSTest extends JasmineTest {
+  describe("BigIntegerConvertTest") {
+    // To test that it works with strict floats, do:
+    //   > set scalaJSSemantics in testSuite ~= { _.withStrictFloats(true) }
+    when("strict-floats").
+      it("testDoubleValueZero") {
```
Comment (sjrd) [inline test-suite/js/src/test/require-jdk8/org/scalajs/testsuite/javalib/math/BigIntegerConvertJSTest.scala]: Don't indent this line (nor the following line). The style for `when`s is:
```scala
    when("strict-floats").
    it("testDoubleValueZero") {
      ...
    }
```

**Comment by sjrd** [comment]
That's all.

**Comment by nicolasstucki** [comment]
`AssertThrows` was renamed to `AssertThrowsUtils` because I started adding util methods on it and not only the backport of the `assertThrows`/`expectThrows`. I specially want to make explicit the existence of new overloads of `expectThrows` which takes a by name parameter.

**Comment by sjrd** [comment]
Hum, OK.

**Comment by nicolasstucki** [comment]
For files that show "No changes" in the first commit it seems to be an issue with github, if I use `git show` on that commit every change appears as it should.

**Comment by sjrd** [comment]
That's all.

**Comment by nicolasstucki** [comment]
Dear Jenkins, retest this please.

**Comment by sjrd** [comment]
LGTM

**Pull request merged**